docs(langsmith): update voice tracing guides for the langsmith.integrations SDK (#4636)
langchain-ai/docs: src/langsmith/trace-with-livekit.mdx

        room=ctx.room,
        agent=Agent(instructions="You are a helpful assistant."),
        record={"audio": True},
    )
```

In console mode, also pass `--record` on the command line. The recording reflects what was played to the client, so a barge-in shows up truncated.

<Warning>
Do not use `audio_path_provider` in production. In a deployed worker, `ctx.session_directory` is an ephemeral temporary directory that LiveKit deletes when the session ends, so there is no durable file to embed.
</Warning>

### Production: record with Egress and attach the file

In production, record the room with [LiveKit Egress](https://docs.livekit.io/home/egress/overview/) into your own object storage, then attach the finished recording to the trace as a real audio attachment. Egress finishes uploading after the call ends, so the integration holds the conversation's root span open until you supply the bytes:

1. Call `processor.expect_recording(thread_id)` when you start egress.
2. After the call, wait for egress to complete, download the file from your storage, and call `processor.complete_recording(thread_id, audio_bytes)`. The integration embeds the bytes and exports the trace.

> You must use the same `thread_id` for expect_recording and complete_recording, so that we can match the recording to the right conversation.

Reviewer: ```suggestion
You must use the same `thread_id` for expect_recording and complete_recording, so the recording is matched to the right conversation.
```

                    ),
                )
            ],
        )
    )
    # Hold the trace open until the recording is ready.
    processor.expect_recording(thread_id)

    async def attach_recording():
        try:
            await wait_for_egress(lkapi, egress.egress_id)  # poll until EGRESS_COMPLETE
            audio = download_from_storage(RECORDING_BUCKET, key)  # your storage client
            processor.complete_recording(thread_id, audio, name="call.ogg")
        except Exception:
            processor.complete_recording(thread_id, None)  # release without audio

    ctx.add_shutdown_callback(attach_recording)

    session = AgentSession(...)
    await session.start(room=ctx.room, agent=Agent(instructions="..."))
```

`wait_for_egress` polls [`list_egress`](https://docs.livekit.io/home/egress/api/) until the status is `EGRESS_COMPLETE` (or subscribe to the `egress_ended` webhook), and `download_from_storage` reads the object with your cloud provider's client. LiveKit Egress also writes to [Google Cloud Storage and Azure](https://docs.livekit.io/home/egress/overview/): swap `s3=` for `gcp=api.GCPUpload(...)` or `azure=api.AzureBlobUpload(...)`.

<Note>

> Because the trace's root span is held until `complete_recording` runs, always call it, including on failure with `data=None`, so the trace is not left open. If the worker stops first, the integration will flush the trace without audio.

Reviewer: ```suggestion
Always call `complete_recording` because the trace's root span is held until it runs, including on failure with `data=None`. If the worker stops first, the integration will flush the trace without audio.
```
